DataFormats/SoATemplate/interface/SoACommon.h

```diff
@@ -32,20 +35,20 @@
 
 // Exception throwing (or willful crash in kernels)
 #if defined(__CUDACC__) && defined(__CUDA_ARCH__)
-#define SOA_THROW_OUT_OF_RANGE(A) \
-  {                               \
-    printf("%s\n", (A));          \
-    __trap();                     \
+#define SOA_THROW_OUT_OF_RANGE(A, I, R)                      \
+  {                                                          \
+    printf("%s: index %d out of range %d\n", (A), (I), (R)); \
+    __trap();                                                \
   }
 #elif defined(__HIPCC__) && defined(__HIP_DEVICE_COMPILE__)
-#define SOA_THROW_OUT_OF_RANGE(A) \
-  {                               \
-    printf("%s\n", (A));          \
-    abort();                      \
+#define SOA_THROW_OUT_OF_RANGE(A, I, R)                      \
+  {                                                          \
+    printf("%s: index %d out of range %d\n", (A), (I), (R)); \
+    abort();                                                 \
   }
 #else
-#define SOA_THROW_OUT_OF_RANGE(A) \
-  { throw std::out_of_range(A); }
+#define SOA_THROW_OUT_OF_RANGE(A, I, R) \
+  { throw std::out_of_range(fmt::format("{}: index {} out of range {}", (A), (I), (R))); }
```

Comment:
@makortel can we use `fmt` in DataFormats ?

Reply:
Yes. While `fmt` is not (yet) being explicitly used in `DataFormats`, it is used in `FWCore/Utilities` and `FWCore/MessageLogger` that both are allowed dependencies in `DataFormats`.

Written that, once we get to gcc 13/14, we will probably start looking into replacing some (if not all) uses of `fmt::format` with `std::format`.

Reply:
👍🏻